server/src/main/java/org/elasticsearch/cluster/coordination/PublicationTransportHandler.java

```diff
@@ -165,6 +167,8 @@ private PublishWithJoinResponse handleIncomingPublishRequest(BytesTransportReque
                     ClusterState incomingState;
                     try {
                         Diff<ClusterState> diff = ClusterState.readDiffFrom(in, lastSeen.nodes().getLocalNode());
+                        // Close early to release resources used by the de-compression as early as possible
+                        in.close();
```

Comment:
Is it that important to close before applying the diff here? Otherwise you could use try with resources

Reply:
Yea we want to get the handle for the `Inflater` before because index metadata operations will use it potentially, but just closing here is pretty lazy admittedly, I'll make it nicer using try-with-resources.